### ISSUE TITLE
kosakuin 類似技術・製品の調査

### ISSUE THREAD

**Issue by key-amb** [aozorahack/kosakuin#6]
- Git ベース
  - GitHub
  - GitLab
  - GitBucket
- コードレビューシステム
  - Review Board

アリモノでプロトタイプとして使えるのであれば、それが手っ取り早いかなという気もしています。

今のところ上に挙げたものは GitHub を除いて OSS なので、kosakuin の一部として使うこともできるかもしれません。

他にこんなものあるよ、とかありましたらお知らせ下さい。

**Comment by cognitom** [comment]
いくつか挙げてみます。
- GitBook: https://www.gitbook.com - [リポジトリ](https://github.com/GitbookIO/gitbook)
- softcover: https://www.softcover.io - [リポジトリ](https://github.com/softcover/softcover)
- Leanpub: https://leanpub.com
- ラフノート(国産): https://ruffnote.com

上記、リポジトリのリンクをつけたものはオープンソースです。このあたりは、 @takahashim さんの方が詳しそう。ラフノートは書籍というよりはドキュメント共有のためのサービスで、バックエンドはGitだと聞いたような気がします。

**Comment by takahashim** [comment]
はい、当然のように詳しいわけですが、挙げていただいているものは「マークアップ言語」や「配布サイト」とかに密結合していているのが気になってます。ので、あんまり気にしなくて良いのではないでしょうか。
GitBookはWebでの表示が良いなあと思うのですが、JS的にはどうなんだろう（もうちょっとシンプルなSPA的にした方がいいのでは）…と思うところもあって、ここはむしろ私よりも最近のJSに詳しそうな @cognitom さんの印象が気になるのでした。

**Comment by cognitom** [comment]
@takahashim さん、コメントありがとうございます。GitBookは一種の静的サイトジェネレータなので、SPAのフレームワーク戦争を避けたのか、Gruntを使っているあたり若干古めな構成なので、開始時点では選びようがなかったのかも。

私も、密結合なのは非常に気になっていて、どれも参考にするにとどめたい印象です。編集ツールと配布サイトは明確に分けたい...。軽やかに行きたいな〜。

「工作院」が受け持つのが、ネットワークというか「配布サイト」側なのだとしたら、Gitクライアント機能も不要ですね。npmやgemのモデルはシンプルでありかなと思います。
- npm: https://www.npmjs.com - [リポジトリ](https://github.com/npm/npm)
- RubyGems: https://rubygems.org
- Packagist: https://packagist.org
